src/libexpr/include/nix/expr/eval-settings.hh

```diff
@@ -193,6 +194,26 @@ struct EvalSettings : Config
           `flamegraph.pl`.
         )"};
 
+    Setting<EvalProfilerMode> evalProfilerMode{this, epmDisabled, "eval-profiler",
+        R"(
+          Enables evaluation profiling. The following modes are supported:
+
+          * `flamegraph` stack sampling profiler. Produces profile suitable for `flamegraph.pl`.
+
+          Use [`eval-profile-file`](#conf-eval-profile-file) to specify where the profile is saved.
+        )"};
+
+    Setting<Path> evalProfileFile{this, "nix.profile", "eval-profile-file",
+        R"(
+          Specifies the file where [evaluation profile](#conf-eval-profiler) is saved.
+        )"};
+
+    Setting<uint32_t> evalProfilerPeriod{this, 10, "eval-profiler-sample-period",
```

Comment:
Both bcc's profile and perf use frequency rather than sample period. I think we should do the same as it is more common.